data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -84,26 +91,39 @@ ReceiveMessageRequest createReceiveMessageRequest() {
     }
 
     private Map<Message, S3EventNotification.S3EventNotificationRecord> getS3MessageEventNotificationRecordMap(final List<Message> sqsMessages) {
-        return sqsMessages.stream().collect(Collectors.toMap(message -> message, this::convertS3EventMessages));
+        return sqsMessages.stream()
+                .collect(HashMap::new,
+                        (messageRecordMap, message) -> messageRecordMap.put(message, convertS3EventMessages(message)),
+                        HashMap::putAll);
     }
 
     S3EventNotification.S3EventNotificationRecord convertS3EventMessages(final Message message) {
-        return S3EventNotification.parseJson(message.body()).getRecords().get(0);
+        try {
+            return S3EventNotification.parseJson(message.body()).getRecords().get(0);
+        } catch (SdkClientException e) {
+            LOG.error("Invalid JSON string in message body of {}", message.messageId());
+        } catch (NullPointerException e) {
+            LOG.error("Unable to convert message body of {} to S3EventNotificationRecord", message.messageId());
+        }
+        return null;
     }
 
     private void processS3ObjectAndDeleteSqsMessages(final Map<Message, S3EventNotification.S3EventNotificationRecord> s3EventNotificationRecords) {
         for (Map.Entry<Message, S3EventNotification.S3EventNotificationRecord> entry: s3EventNotificationRecords.entrySet()) {
-            if(isEventNameCreated(entry.getValue())) {
-                S3ObjectReference s3ObjectReference = populateS3Reference(entry.getValue());
+            if (entry.getValue() == null) {
+                // TODO: delete sqsMessages whose S3EventNotificationRecord is null
```

Comment:
Do we want to delete messages we couldn't process or should they go to DLQ?

Reply:
We could add a configuration to check if DLQ is enabled.If enabled we ignore the message and let DLQ handle it. It not enabled we will delete the message.